hibernate-core/src/main/java/org/hibernate/proxy/pojo/ProxyFactoryHelper.java

```diff
@@ -84,14 +85,21 @@ public static void validateProxyability(final PersistentClass persistentClass) {
 		Class clazz = persistentClass.getMappedClass();
 		while ( properties.hasNext() ) {
 			Property property = (Property) properties.next();
-			Method method = property.getGetter( clazz ).getMethod();
-			if ( method != null && Modifier.isFinal( method.getModifiers() ) ) {
-				LOG.gettersOfLazyClassesCannotBeFinal( persistentClass.getEntityName(), property.getName() );
-			}
-			method = property.getSetter( clazz ).getMethod();
-			if ( method != null && Modifier.isFinal( method.getModifiers() ) ) {
-				LOG.settersOfLazyClassesCannotBeFinal( persistentClass.getEntityName(), property.getName() );
-			}
+			validateGetterSetterMethodProxyability( "Getter", property.getGetter( clazz ).getMethod() );
+			validateGetterSetterMethodProxyability( "Setter", property.getSetter( clazz ).getMethod() );
+		}
+	}
+
+	public static void validateGetterSetterMethodProxyability(String getterOrSetter, Method method ) {
+		if ( method != null && Modifier.isFinal( method.getModifiers() ) ) {
+			throw new HibernateException(
+					String.format(
+							"%s methods of lazy classes cannot be final: %s#%s",
+							getterOrSetter,
+							method.getDeclaringClass().getName(),
+							method.getName()
+					)
+			);
```

Comment:
It seems `getterOrSetter` parameter is not necessary for we can get that information if we refactor the second parameter to accept either `Getter` or `Setter`. For instance
```
private static void validateGetterSetterMethodProxyability(Object getterOrSetter ) {
                String methodCategory = getterOrSetter instanceof Getter ? "Getter" : "Setter";
                Method method = getterOrSetter instanceof Getter ? Getter.class.cast(getterOrSetter).getMethod)(): Setter.class.cast(getterOrSetter).getMethod();
		if ( method != null && Modifier.isFinal( method.getModifiers() ) ) {
			throw new HibernateException(
					String.format(
							"%s methods of lazy classes cannot be final: %s#%s",
							methodCategory,
							method.getDeclaringClass().getName(),
							method.getName()
					)
			);
```
I also changed the visibility from public to private above. If `Getter` and `Setter` shared some common interface (like `methodAware`), it would be more elegant and straightforward.
not a big deal for sure and feel free to ignore my comment.

Reply:
@NathanQingyangXu , `PojoEntityTuplizer#buildProxyFactory` also calls `ProxyFactoryHelper#validateGetterSetterMethodProxyability```, so it needs to be public.

Of course, `ProxyFactoryHelper#validateProxyability` could have repeated the call to `ProxyFactoryHelper#extractProxyGetIdentifierMethod`, but then I would have to change the signature of the method to accept a `Setter` and a `Getter`. 

Since `ProxyFactoryHelper` is an SPI, I would need to deprecate the original method. It just didn't seem worthwhile.

I just did what I thought would have the smallest impact.

Reply:
thanks for the explanation